### PR TITLE
RHOAIENG-730: RStudio Server notebook images

### PR DESCRIPTION
**Pull request by grainnejenningsRH**
Adds entries for the RStudio Server and CUDA - RStudio Server notebook images in modules/notebook-images-for-data-scientists.adoc and modules/options-for-notebook-server-environments.adoc. 

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by jstourac — APPROVED** [review]

**Review by jiridanek — APPROVED** [review]

**Review by jstourac — APPROVED** [review]

File: modules/configuring-your-ide.adoc [hunk 1]
```diff
@@ -63,3 +63,60 @@ The extension you installed appears in the *Browser - Installed* list on the *Ex
 
 See link:https://open-vsx.org/[Open VSX Registry] for available third-party extensions that you can consider installing.
 
+ifndef::upstream[]
+== Building the RStudio Server notebook images
+
+To use the *RStudio Server* and *CUDA - RStudio Server* notebook images, you must first build them by creating a secret and triggering the BuildConfig, and then enable them in the {productname-short} UI by editing the `rstudio-rhel9` and `cuda-rstudio-rhel9` imagestreams.
```
Comment (chtyler) [inline modules/configuring-your-ide.adoc:69]: See: https://docs.openshift.com/container-platform/4.14/openshift_images/index.html

I think it should be "image streams".

File: modules/configuring-your-ide.adoc [hunk 2]
```diff
@@ -63,3 +63,60 @@ The extension you installed appears in the *Browser - Installed* list on the *Ex
...
+.Prerequisites
+
+* Before starting the RStudio Server build process, you have at least 1 CPU and 2Gi memory available for `rstudio-server-rhel9`, and 1.5 CPUs and 8Gi memory available for `cuda-rstudio-server-rhel9` on your cluster.
+* You are logged in to your OpenShift cluster.
+* You have the `cluster-admin` role in {openshift-platform}, to the namespace `rhoai-applications`, or with cluster-wide role binding.
+* You have an active {org-name} Enterprise Linux (RHEL) subscription.
+
+.Procedure
+
+. Create a secret with subscription manager credentials. These are usually your {org-name} Customer Portal username and password.
```
Comment (chtyler) [inline modules/configuring-your-ide.adoc:80]: I've seen this referred to as "Red Hat Subscription Manager", "Subscription Manager", "Red Hat Subscription Management". I'd be tempted to capitalize it at least.

File: modules/configuring-your-ide.adoc [hunk 3]
```diff
@@ -63,3 +63,60 @@ The extension you installed appears in the *Browser - Installed* list on the *Ex
...
++
+Note: The secret must be named `rhel-subscription-secret`, and the keys `USERNAME` and `PASSWORD` within it must be in capital letters.
```
Comment (chtyler) [inline modules/configuring-your-ide.adoc:82]: could also word this as "... and its `USERNAME` and `PASSWORD` keys..."

File: modules/configuring-your-ide.adoc [hunk 4]
```diff
@@ -63,3 +63,60 @@ The extension you installed appears in the *Browser - Installed* list on the *Ex
 
 See link:https://open-vsx.org/[Open VSX Registry] for available third-party extensions that you can consider installing.
 
+ifndef::upstream[]
+== Building the RStudio Server notebook images
+
+To use the *RStudio Server* and *CUDA - RStudio Server* notebook images, you must first build them by creating a secret and triggering the BuildConfig, and then enable them in the {productname-short} UI by editing the `rstudio-rhel9` and `cuda-rstudio-rhel9` imagestreams.
+
+.Prerequisites
+
```
Comment (chtyler) [inline modules/configuring-your-ide.adoc:72]: The spaces after .Prerequisites, .Procedure, and .Verification are not required.

File: modules/notebook-images-for-data-scientists.adoc [hunk 5]
```diff
@@ -78,6 +78,21 @@ For more information about the support scope of Red{nbsp}Hat Technology Preview
...
+endif::[]
+
+| CUDA - RStudio Server
+| Use the CUDA - RStudio Server notebook image to access the RStudio IDE and NVIDIA CUDA Toolkit. RStudio is an integrated development environment for R, a programming language for statistical computing and graphics. With the NVIDIA CUDA toolkit, you can optimize your work using GPU-accelerated libraries and optimization tools.
+See link:https://posit.co/products/open-source/rstudio-server/[the RStudio Server site] for more information. +
+
+ifndef::upstream[]
+NOTE: To use the *CUDA - RStudio Server* notebook image, you must first build it by creating a secret and triggering the BuildConfig, and then enable it in the {productname-short} UI by editing the `cuda-rstudio-rhel9` imagestreams. See link:{rhoaidocshome}{default-format-url}/getting_started_with_{url-productname-long}/configuring-your-ide_get-started#building_the_rstudio_server_notebook_images[Building the RStudio Server notebook images] for more information.
```
Comment (chtyler) [inline modules/notebook-images-for-data-scientists.adoc:94]: See previous comment about image streams.

File: modules/configuring-your-ide.adoc (hunk 2 above)
Comment (jiridanek) [reply modules/configuring-your-ide.adoc:80]: or take it as commend name and go with `subscription-manager`

File: modules/configuring-your-ide.adoc (hunk 1 above)
Comment (grainnejenningsRH) [reply modules/configuring-your-ide.adoc]: Good shout, updated.

File: modules/configuring-your-ide.adoc (hunk 2 above)
Comment (grainnejenningsRH) [reply modules/configuring-your-ide.adoc]: Have capitalised. Thanks!

File: modules/configuring-your-ide.adoc (hunk 3 above)
Comment (grainnejenningsRH) [reply modules/configuring-your-ide.adoc]: Good suggestion, have updated.

File: modules/notebook-images-for-data-scientists.adoc (hunk 5 above)
Comment (grainnejenningsRH) [reply modules/notebook-images-for-data-scientists.adoc]: Updated.

**Review by chtyler — APPROVED** [review]

**Pull request merged**